src/assets/scripts/client/math/flightMath.js

```diff
@@ -203,8 +203,9 @@ export const calculateTurnInitiaionDistance = (aircraft, currentWaypointPosition
     const nominalBankAngleDegrees = 25;
     const speed = kn_ms(aircraft.speed);
     const bankAngle = degreesToRadians(nominalBankAngleDegrees);
+    const nextWaypointModel = aircraft.fms.currentLeg.nextWaypoint;
```

Comment:
does this still work when the `currentWaypoint` _is_ the last waypoint in a leg?  

`aircraft.fms.getNextWaypointPositionModel()` 
- will check for that
- skip to the next leg and check for a waypoint 
- and only return the position.